packages/engine-core/src/data-proxy/DataProxyEngine.ts

```diff
@@ -151,13 +160,19 @@ export class DataProxyEngine extends Engine {
         .join('\n')}`,
     })
 
-    const body: QueryEngineBatchRequest = {
-      batch: queries.map(({ query }) => ({ query, variables: {} })),
-      transaction: isTransaction,
-      isolationLevel: transaction?.isolationLevel,
+    const headers: Record<string, string> = { ...customDataProxyHeaders }
+    if (traceparent) {
+      headers.traceparent = traceparent
+    }
+
+    const itx = transaction?.kind === 'itx' ? transaction.options : undefined
+    if (itx) {
+      headers['X-transaction-id'] = itx.id
```

Comment:
Do you think it would be possible to factor/deduplicate this shared logic into `requestInternal`?

Reply:
Looks similar to https://github.com/prisma/prisma/pull/17540/files#diff-b4fc1a3eee6a1f0140a901585106e8b74dfc77aff2f46bb68e5cd502450dabf5R139

Reply:
Done